server/app/graphql/mutations/delete_user.rb

```diff
@@ -0,0 +1,25 @@
+module Mutations
+  class DeleteUser < BaseMutation
+    # TODO: define return fields
+    # field :post, Types::PostType, null: false
+
+    # TODO: define arguments
+    # argument :name, String, required: true
+
+    # TODO: define resolve method
+    # def resolve(name:)
+    #   { post: ... }
+    # end
+    field :user, Types::UserType, null: false
+    argument :name, String, required: true
+
+    def resolve(name:)
+      user = User.find_by(name: name)
+      if !user || !user.destroy
+        raise GraphQL::ExecutionError, "Failed to delete user"
+      else
+        { user: user }
+      end
```

Comment:
これってUserが存在しないケースって考えられないのでは？
また，削除成功時にuserを返すってのは違和感を感じる。普通に何も返さなくて良さそう。
```suggestion
      user = User.find_by!(name: name)
      if !user.destroy
        raise GraphQL::ExecutionError, "Failed to delete user"
      end
```